smart-contracts/contracts/mixins/MixinKeys.sol

```diff
@@ -654,41 +660,39 @@ contract MixinKeys is
   }
 
   /**
-   * @notice Change the maximum number of keys the lock can edit
+   * Set the maximum number of keys a specific address can use
+   * A function to change the default duration of each key in the lock
+   * - Change the maximum number of keys the lock can edit
+   * @notice keys previously bought are unaffected by this change in expiration duration (i.e.
+   * existing keys timestamps are not recalculated/updated)
+   * @param _newExpirationDuration the new amount of time for each key purchased or type(uint).max for a non-expiring key
+   * @param _maxKeysPerAcccount the maximum amount of key a single user can own
    * @param _maxNumberOfKeys uint the maximum number of keys
-   * @dev Can't be smaller than the existing supply
+   * @dev _maxNumberOfKeys Can't be smaller than the existing supply 
    */
-  function setMaxNumberOfKeys (uint _maxNumberOfKeys) external {
+   function updateLockConfig(
+    uint _newExpirationDuration,
+    uint _maxKeysPerAcccount,
+    uint _maxNumberOfKeys
+  ) external {
      _onlyLockManager();
+     if(_maxKeysPerAcccount == 0) {
```

Comment:
Is that an issue though? I am not sure... It probably means nobody can purchase keys after this?

Reply:
We can maybe save some size by removing it

Reply:
We can keep that for now as it wont be significant in terms of size but will make the config step more error prone